### PR TITLE
Adding mocks and unit tests for socket code

### PR DESCRIPTION
**Pull request by nsarras**

### CODE REVIEW

**Comment by nsarras** [comment]
Pycode style issues need to be resolved 

dss_datamover/socket_communication.py:404:1: W391 blank line at end of file
dss_datamover/tests/conftest.py:112:33: E225 missing whitespace around operator
dss_datamover/tests/conftest.py:126:28: E261 at least two spaces before inline comment
dss_datamover/tests/conftest.py:127:30: E261 at least two spaces before inline comment
dss_datamover/tests/conftest.py:149:44: E203 whitespace before ':'
dss_datamover/tests/conftest.py:160:48: E203 whitespace before ':'
dss_datamover/tests/conftest.py:160:50: E202 whitespace before ']'
dss_datamover/tests/conftest.py:163:48: E203 whitespace before ':'
dss_datamover/tests/conftest.py:196:1: W293 blank line contains whitespace
dss_datamover/tests/conftest.py:205:1: W293 blank line contains whitespace
dss_datamover/tests/conftest.py:209:1: E302 expected 2 blank lines, found 1
dss_datamover/tests/conftest.py:217:29: E231 missing whitespace after ':'
dss_datamover/tests/conftest.py:221:1: W293 blank line contains whitespace
dss_datamover/tests/conftest.py:224:1: W293 blank line contains whitespace
dss_datamover/tests/conftest.py:227:1: W293 blank line contains whitespace
dss_datamover/tests/conftest.py:246:1: E302 expected 2 blank lines, found 1
dss_datamover/tests/conftest.py:308:1: E302 expected 2 blank lines, found 1
dss_datamover/tests/conftest.py:311:1: W293 blank line contains whitespace
dss_datamover/tests/conftest.py:321:1: W293 blank line contains whitespace
dss_datamover/tests/conftest.py:326:1: W293 blank line contains whitespace
dss_datamover/tests/test_socket.py:44:1: E302 expected 2 blank lines, found 1
dss_datamover/tests/test_socket.py:46:32: E226 missing whitespace around arithmetic operator
dss_datamover/tests/test_socket.py:52:1: E302 expected 2 blank lines, found 1
dss_datamover/tests/test_socket.py:58:1: W293 blank line contains whitespace
dss_datamover/tests/test_socket.py:65:1: W293 blank line contains whitespace
dss_datamover/tests/test_socket.py:75:20: E712 comparison to True should be 'if cond is True:' or 'if cond:'
dss_datamover/tests/test_socket.py:78:20: E712 comparison to True should be 'if cond is True:' or 'if cond:'
dss_datamover/tests/test_socket.py:84:1: W293 blank line contains whitespace
dss_datamover/tests/test_socket.py:86:20: E712 comparison to False should be 'if cond is False:' or 'if not cond:'
dss_datamover/tests/test_socket.py:87:1: W293 blank line contains whitespace
dss_datamover/tests/test_socket.py:95:48: E231 missing whitespace after ':'
dss_datamover/tests/test_socket.py:96:24: E712 comparison to False should be 'if cond is False:' or 'if not cond:'
dss_datamover/tests/test_socket.py:97:1: W293 blank line contains whitespace
dss_datamover/tests/test_socket.py:104:20: E712 comparison to True should be 'if cond is True:' or 'if cond:'
dss_datamover/tests/test_socket.py:117:1: W293 blank line contains whitespace
dss_datamover/tests/test_socket.py:121:1: W293 blank line contains whitespace
dss_datamover/tests/test_socket.py:139:60: E226 missing whitespace around arithmetic operator
dss_datamover/tests/test_socket.py:225:1: W293 blank line contains whitespace
dss_datamover/tests/test_socket.py:229:1: W293 blank line contains whitespace
dss_datamover/tests/test_socket.py:235:1: W293 blank line contains whitespace
dss_datamover/tests/test_socket.py:238:1: W293 blank line contains whitespace
dss_datamover/tests/test_socket.py:244:20: E712 comparison to False should be 'if cond is False:' or 'if not cond:'
dss_datamover/tests/test_socket.py:249:20: E712 comparison to True should be 'if cond is True:' or 'if cond:'
dss_datamover/tests/test_socket.py:251:20: E712 comparison to True should be 'if cond is True:' or 'if cond:'
dss_datamover/tests/test_socket.py:258:1: W293 blank line contains whitespace
dss_datamover/tests/test_socket.py:260:20: E712 comparison to False should be 'if cond is False:' or 'if not cond:'
dss_datamover/tests/test_socket.py:261:1: W293 blank line contains whitespace
dss_datamover/tests/test_socket.py:269:44: E231 missing whitespace after ':'
dss_datamover/tests/test_socket.py:270:20: E712 comparison to False should be 'if cond is False:' or 'if not cond:'
dss_datamover/tests/test_socket.py:272:1: W293 blank line contains whitespace
dss_datamover/tests/test_socket.py:280:20: E712 comparison to True should be 'if cond is True:' or 'if cond:'
dss_datamover/tests/test_socket.py:297:1: W293 blank line contains whitespace
dss_datamover/tests/test_socket.py:317:60: E226 missing whitespace around arithmetic operator

File: dss_datamover/tests/conftest.py [hunk 1]
```diff
@@ -103,20 +107,141 @@ class MockSocket():
...
+    STATUS_MISALIGNEDBUFSIZE = 6
+    STATUS_WRONGBUFSIZE = 7
+    STATUS_LISTENING = 8
+    STATUS_CLOSED = 9
 
-    def connect(self):
-        return True
-
-    def recv(self):
+    # TODO: finish building out MockSocket class
```
Comment (nsarras) [inline dss_datamover/tests/conftest.py:121]: remove this comment

File: dss_datamover/tests/conftest.py [hunk 2]
```diff
@@ -103,20 +107,141 @@ class MockSocket():
     """
     Dummy Object for an actual socket, should simulate all basic functions of a socket object
     """
-    # TODO: finish building out MockSocket class
-    def __init__(self):
-        self.host = "xxx.xxxx.xxxx.xxxx"
-        self.port = "xxxx"
+    STATUS_NORMAL = 0
```
Comment (nsarras) [inline dss_datamover/tests/conftest.py:110]: these DEFAULTS should be near the top of the file

File: dss_datamover/tests/conftest.py [hunk 3]
```diff
@@ -179,3 +304,34 @@ def _method(master):
...
+class MockMinio():
+    def __init__(self):
+        self.data = {}
+    
+    def list(self, key=''):
+        if not key:
+            return list(self.data.items())
+        else:
+            ret = []
+            for k in self.data.keys():
```
Comment (nsarras) [inline dss_datamover/tests/conftest.py:317]: the specific key should only show up once. You can check to see if it is a nested dict, if so, just the items() method will pack the dict into a list of tuples

`self.data[k].items() if isinstance(self.data[k], dict) else [(k, self.data[k])]`

File: dss_datamover/tests/test_socket.py [hunk 4]
```diff
@@ -33,8 +33,362 @@
...
+    if wrong_size:
+        msg_len = (str(len(msg)+3)).zfill(header_len)
+    else:
+        msg_len = (str(len(msg))).zfill(header_len)
+    ret = msg_len + msg
+    return ret
 
-@pytest.mark.usefixtures("get_mock_clientsocket")
+@pytest.mark.usefixtures("get_pytest_configs", "get_config_dict", "get_mock_logger", "get_header_length")
 class TestSocketCommunication():
```
Comment (nsarras) [inline dss_datamover/tests/test_socket.py:53]: Great job on the tests, only recommendation I have is to consolidate them more, meaning that if there are tests that test the positive & negative & corner cases of the same funcitonality, such as socket connect. Then all of those variations should be within that same test case

File: dss_datamover/socket_communication.py [hunk 5]
```diff
@@ -55,7 +55,7 @@ def __init__(self, config, logger=None):
         self.logger = logger
         self.config = config
         self.socket = None
-        self.last_exception_log_time = datetime.now()
+        self.last_exception_log_time = datetime.min
```
Comment (nsarras) [inline dss_datamover/socket_communication.py:58]: change is not necessary please remove

File: dss_datamover/socket_communication.py [hunk 6]
```diff
@@ -232,7 +232,7 @@ def __init__(self, config, logger=None):
         self.logger = logger
         self.config = config
         self.client_socket = None
-        self.last_exception_log_time = datetime.now()
+        self.last_exception_log_time = datetime.min
```
Comment (nsarras) [inline dss_datamover/socket_communication.py:235]: change is not necessary please remove

**Comment by nsarras** [comment]
Pycode style issue needs to be resolved ; dss_datamover/socket_communication.py:404:1: W391 blank line at end of file

**Comment by nsarras** [comment]
Sonar Cloud security hotspot dss_datamover/tests/conftest.py line 209. Make sure ip address is safe

**Review by grandsuri — APPROVED** [review]

**Comment by nsarras** [comment]
Eric, please recreate PR under your own user and pull in master branch

**Comment by nsarras** [comment]
closing, as Eric has included changes in another PR
